core/kernel/transfer_list.c

```diff
@@ -389,13 +389,16 @@ bool transfer_list_set_data_size(struct transfer_list_header *tl,
 		    tl->size + mov_dis > tl->max_size) {
 			return false;
 		}
-		ru_new_ev = old_ev + mov_dis;
-		memmove((void *)ru_new_ev, (void *)old_ev, tl_old_ev - old_ev);
+		r_new_ev = old_ev + mov_dis;
 		tl->size += mov_dis;
-		gap = ru_new_ev - new_ev;
 	} else {
-		gap = old_ev - new_ev;
+		mov_dis = ROUNDDOWN(old_ev - new_ev,
+				    TL_ALIGNMENT_FROM_ORDER(tl->alignment));
+		r_new_ev = old_ev - mov_dis;
+		tl->size -= mov_dis;
 	}
+	gap = r_new_ev - new_ev;
+	memmove((void *)r_new_ev, (void *)old_ev, tl_old_ev - old_ev);
 
 	if (gap >= sizeof(*dummy_te)) {
 		/* Create a dummy transfer entry to fill up the gap */
```

Comment:
I think we still need it, if the alignment is big enough that the gap will have a chance to bigger than a TE header.

Reply:
What about this passage in the commit message?
> Releasing the unused bytes from transfer list when downsizing the data
of a transfer entry is required, since currently allocating new transfer
entries from spaces marked with 'TL_TAG_EMPTY' is not supported.